system/security/shared/caching.cpp

```diff
@@ -711,8 +741,16 @@ void CPermissionsCache::flush()
             delete (*ui).second;
         m_userCache.clear();
     }
+    if (m_useLegacyDefaultFileScopePermissionCache)
+    {
+        m_defaultPermission = SecAccess_None;
```

Comment:
should this be :
m_defaultPermission = SecAccess_Unknown

.. to trigger the next call to queryDefaultPermission to get new permissions?

Reply:
Yes, good catch.